client/layout/index.jsx

```diff
@@ -175,9 +175,10 @@ export default connect( state => {
 	const sectionGroup = getSectionGroup( state );
 	const sectionName = getSectionName( state );
 	const currentRoute = getCurrentRoute( state );
+	const noMasterbarForSection = 'signup' === sectionName || 'jetpack-connect' === sectionName;
 
 	return {
-		masterbarIsHidden: ! masterbarIsVisible( state ) || 'signup' === sectionName,
+		masterbarIsHidden: ! masterbarIsVisible( state ) || noMasterbarForSection,
```

Comment:
Should this logic go inside the `masterbarIsVisible` selector?

Reply:
See https://github.com/Automattic/wp-calypso/pull/30958#discussion_r263781783